File: simulation/TrueBattleBotSim/Assets/Scripts/Scenarios/SequenceElementConfig.cs
using System;

[Serializable]
public class SequenceElementConfig : ICloneable
{
    public float timestamp = 0.0f;

    // Waypoint follower fields
    public float x = 0.0f;  // meters
    public float y = 0.0f;  // meters
    public float z = 0.0f;  // meters
    public float roll = 0.0f;  // degrees
    public float pitch = 0.0f;  // degrees
    public float yaw = 0.0f;  // degrees
    public float vx = 0.0f;  // meters/sec
    public float vy = 0.0f;  // meters/sec
    public float vz = 0.0f;  // meters/sec
    public float vroll = 0.0f;  // degrees/sec
    public float vpitch = 0.0f;  // degrees/sec
    public float vyaw = 0.0f;  // degrees/sec

    // Target follower fields
    public string target_name = "";
    public string secondary_target_name = "";

    // Loop back to the top
    public bool reset = false;

    public object Clone()
    {
        return MemberwiseClone();
    }
}

File: shared/bw_shared/simulation_control/configs/sequence_element_config.py
from dataclasses import dataclass

from bw_shared.simulation_control.enums.actor_role import ActorRole


@dataclass
class SequenceElementConfig:
    timestamp: float = 0.0

    x: float = 0.0
    y: float = 0.0
    z: float = 0.0
    roll: float = 0.0
    pitch: float = 0.0
    yaw: float = 0.0
    vx: float = 0.0
    vy: float = 0.0
    vz: float = 0.0
    vroll: float = 0.0
    vpitch: float = 0.0
    vyaw: float = 0.0

    target_name: ActorRole = ActorRole.EMPTY
    secondary_target_name: ActorRole = ActorRole.EMPTY

    reset: bool = False
